parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+# import random
+import threading
+import pickle
+import time
+import queue
+import uuid
+import zmq
+
+import multiprocessing
+
+from ipyparallel.serialize import unpack_apply_message  # pack_apply_message,
+from ipyparallel.serialize import serialize_object
+
+RESULT_TAG = 10
+TASK_REQUEST_TAG = 11
+
+LOOP_SLOWDOWN = 0.0  # in seconds
+
+
+class Manager(object):
+    """ Manager (feudal lord) rules over the workers
+
+    1. Asynchronously queue large volume of tasks
+    2. Allow for workers to join and leave the union
+    3. Detect workers that have failed using heartbeats
+    4. Service single and batch requests from workers
+    5. Be aware of requests worker resource capacity,
+       eg. schedule only jobs that fit into walltime.
+    """
+    def __init__(self,
+                 task_q_url="tcp://127.0.0.1:50097",
+                 result_q_url="tcp://127.0.0.1:50098",
+                 max_queue_size=10,
+                 worker_count=0,
+                 uid=None,
+                 heartbeat_period=30):
+        """
+        Parameters
+        ----------
+        worker_url : str
+             Worker url on which workers will attempt to connect back
+
+        worker_count : int,
```

Comment:
i prefer `None` to `0` to make it more explicit it is a special case

Reply:
Why not `cores` ?

Reply:
agreed!

Reply:
On second thought, maybe `workers_per_core` would be clearer? With a `None` default, where if `workers_per_core=None`, we would set it to num_cores?